hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperTask.java

```diff
@@ -220,27 +311,34 @@ private void  writeOMKeyToContainerDB(String key, OmKeyInfo omKeyInfo)
         ContainerKeyPrefix containerKeyPrefix = new ContainerKeyPrefix(
             containerId, key, keyVersion);
         if (reconContainerMetadataManager.getCountForContainerKeyPrefix(
-            containerKeyPrefix) == 0) {
+            containerKeyPrefix) == 0
+            && !containerKeyMap.containsKey(containerKeyPrefix)) {
           // Save on writes. No need to save same container-key prefix
           // mapping again.
-          reconContainerMetadataManager.storeContainerKeyMapping(
-              containerKeyPrefix, 1);
+          containerKeyMap.put(containerKeyPrefix, 1);
```

Comment:
Will the value of each entry in `containerKeyMap` always be `1`? Will there be other use cases in the future?

Maybe I didn't fully understand this, but if it is just for checking the key existence, wouldn't a `Set` be more memory-efficient?

Reply:
This is something that I didn't fully understand. I didn't want to change that much in the code and I hardly find any information what we store here. What I found is we map the (container, key) -> count here. We use the `ContainerKeyPrefix` as the key, which encapsulates the Key information needed for the Recon container DB. From the javadoc: "Currently, it is the containerId and the whole key + key version." I don't know if we could have `ContainerKeyPrefix`s in a cluster. But in the `TestReconContainerMetadataManagerImpl` test class we add `ContainerKeyPrefix`s with 2 and 3 as counts. I don't know if I should change something here, previously it only added with 1 as counts when we were processing the events.

Reply:
Got it. I'm fine with keeping this as-is. Thx!